core/templates/dev/head/directives.js

```diff
@@ -156,3 +156,17 @@ oppia.directive('mobileFriendlyTooltip', ['$timeout', function($timeout) {
     }
   };
 }]);
+
+oppia.directive('updateLanguage', ['$rootScope', function($rootScope) {
```

Comment:
Yes, but every time i change the language on preference page , until i refresh ,the global variable wont update . Also in preference.js the language will dynamically update . So in order to get that variable in base.js i need to write extra code .
I will try to look at it again.